### DISCUSSION TITLE
Getting `Cannot read properties of undefined (reading 'init')` while clicking on connect button

### DISCUSSION THREAD

**Discussion by MahendraBishnoi29** [WalletConnect/walletconnect-monorepo#2546] [Developer Support]
Hi WalletConnect Team, I'm using Wallet connect for the first time and having tough times with it.

this is my component where i want implement wallet connect's `signIn auth with Next.js` and `signApi`(after connecting the wallet) but when I click on it nothing happens and I get a `error` in console:

![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/74294202/9807d471-7cd1-4d33-87a3-9eb63c30f9fb)

**Here's my Setup:**
![code](https://github.com/WalletConnect/walletconnect-monorepo/assets/74294202/a6ca3124-7dc2-4a8d-8ee8-ea9143e3f0b6)

**here I'm using that function:**
`<WalletButton
                    onClick={onConnect}
                    text="Wallet Connect"
                    src={WalletConnectIcon}
                  />`
                  
                  
**My Root Config setup:**
![1](https://github.com/WalletConnect/walletconnect-monorepo/assets/74294202/3185c2c4-06bb-4755-825c-82cd40a71b23)

              

When i click on the button I want to open the wallet Connect modal and i need the Metamask Option like the pic below with desktop wallets:
<img width="354" alt="image" src="https://github.com/WalletConnect/walletconnect-monorepo/assets/74294202/008d9a24-2939-4b2e-b480-ab867490114b">

Please Help me out.

**Comment by finessevanes** [comment]
@MahendraBishnoi29 apologies you're having a hard time implementing our SDKs. I'm not sure I understand what you're trying to build

> i want implement wallet connect's signIn auth with Next.js 

Are you trying to implement Auth or Sign?

**Reply by MahendraBishnoi29** [reply]
I want to do both @finessevanes after connecting to the user I will be giving a pop-up for sign in my Dapp
